### ISSUE TITLE
False Positive abstract-method for collections.Counter (raising NotImplementedError)

### ISSUE THREAD

**Issue by brycepg** [PyCQA/pylint#1758]
### Steps to reproduce
run pylint on:
```python
import collections
class F(collections.Counter):
    pass
```

collections.Counter has a `fromkeys` which always raises `NotImplementedError` with the warning "Counter.fromkeys() is undefined.  Use Counter(iterable) instead.".

Raising NotImplementedError does not always mean that the method is an abstract method.

However this class violates the Likov-Substitution principle (inherits from dict and then breaks the interface with this exception), so maybe I should just write a checker to suppress this error in stdlib.

### Current behavior

W:  2, 0: Method 'fromkeys' is abstract in class 'Counter' but is not overridden (abstract-method)

### Expected behavior

No warning

### pylint --version output

pylint 1.7.4, 
astroid 1.5.3
Python 3.6.2 |Continuum Analytics, Inc.| (default, Jul 20 2017, 13:51:32) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]

**Comment by PCManticore** [comment]
Yeah, this behaviour of `Counter` feels weird to me. We've been assuming that raising `NotImplementedError` intends to convey that a particular class needs to have those methods overridden or that it is a poor man's abstract class and I would not want to remove this behaviour just for the sake of `Counter`. I would say that this error should be brought up to CPython devs and that the method should be removed rather than raising a NotImplementedError for no reason.
